### ISSUE TITLE
Upgrade to Log4j 2

### ISSUE THREAD

**Issue by demiankatz** [solrmarc/solrmarc#96]
Log4j 1 is well past end of life; we should upgrade to Log4j 2. As an interim solution, the bridge adapter described at https://logging.apache.org/log4j/2.x/log4j-1.2-api/index.html may help.

**Comment by acdha** [comment]
I strongly agree with the decision to upgrade to log4j 2. I had a partial branch in progress but ran out of time to work on it and ended up “cheating” by using reload4j which is a fork of log4j 1 which closes the half dozen CVEs which currently show up on our SolrMarc build:

https://reload4j.qos.ch

This ended up being really simple: have Maven download reload4j-1.2.24.jar and modifying our Dockerfile to remove the bundled log4j-1.2.17.jar.

**Comment by demiankatz** [comment]
@haschart, do you think it might be feasible to issue a new release with @acdha's suggestion for an incremental step forward, and then perhaps we can work on a larger project to get to log4j 2?

@acdha, if you have work in progress on the log4j 2 upgrade, even if it's not finished, perhaps it's worth sharing a draft PR to enable collaboration around this.

**Comment by demiankatz** [comment]
Thanks, @haschart. As always, let me know if I (or the VuFind community generally) can do anything to help!

**Comment by acdha** [comment]
> @acdha, if you have work in progress on the log4j 2 upgrade, even if it's not finished, perhaps it's worth sharing a draft PR to enable collaboration around this.

Thanks, I was hoping to do that but need to detangle the internal fork which I inherited first since a lot of the changes are in code we wrote.

**Comment by demiankatz** [comment]
I've opened #100 with @acdha's suggestion to see if I can help this process along a little bit! :-)
